modules/core/04-channel/keeper/grpc_query.go

```diff
@@ -480,14 +480,25 @@ func (q Keeper) NextSequenceReceive(c context.Context, req *types.QueryNextSeque
 	}
 
 	ctx := sdk.UnwrapSDKContext(c)
-	sequence, found := q.GetNextSequenceRecv(ctx, req.PortId, req.ChannelId)
+	channel, found := q.GetChannel(ctx, req.PortId, req.ChannelId)
 	if !found {
 		return nil, status.Error(
 			codes.NotFound,
-			errorsmod.Wrapf(types.ErrSequenceReceiveNotFound, "port-id: %s, channel-id %s", req.PortId, req.ChannelId).Error(),
+			errorsmod.Wrapf(types.ErrChannelNotFound, "port-id: %s, channel-id %s", req.PortId, req.ChannelId).Error(),
 		)
 	}
 
+	// Return the next sequence received for ordered channels and 0 for unordered channels.
+	var sequence uint64
+	if channel.Ordering == types.ORDERED {
+		sequence, found = q.GetNextSequenceRecv(ctx, req.PortId, req.ChannelId)
```

Comment:
Thanks @DimitrisJim! Code looks great! 

Just want to note that if we add another channel ordering, like `ORDERED_ALLOW_TIMEOUT` we may need to adjust this handling. It's not incredibly clear whether `ORDERED` channels or `UNORDERED` channels are the exception here, but since we are referring to the next sequence receive, it probably makes sense to add an if statement for the channel types which do not use that value (UNORDERED). I see the referenced issue suggested this structure of code.

An alternative solution:
```go
if channel.Ordering == types.UNORDERED {
    // unordered channels do not make use of the next sequence receive
    return 0, nil
}

sequence, found := q.GetNextSequenceRecv(ctx, req.PortId, req.ChannelId)

// etc
```

In this scenario, since we perform the query by default, `ORDERED_ALLOW_TIMEOUT` would function properly without additional changes. I believe we will need to modify this code with the addition of ordered allow timeout channels (not yet implemented)

Reply:
Yep, can totally see that point, wasn't aware of the possibility of another ordered variant. Does it make sense to fix it pronto? (Note that some grep-ing found one other case where we special case on `ORDERED` during [packet acknowledgement](https://github.com/cosmos/ibc-go/blob/ae27fa5960d523e4476776308ecd0b4cd3377671/modules/core/04-channel/keeper/packet.go#L442) but I'm unsure if the same semantics would apply).

Reply:
Yea, the new ordered variant is a bit of a new concept for us as well, which is why this issue wasn't so obvious. I only barely made the connection thinking about explicit return values (I prefer the code to be explicit with what it returns rather than being implicit by not executing some conditional)

I think it could make sense to fix since we know it will be an issue? Might save someone some time debugging later. I'm also perfectly happy having an issue opened. 

> (Note that some grep-ing found one other case where we special case on ORDERED during [packet acknowledgement](https://github.com/cosmos/ibc-go/blob/ae27fa5960d523e4476776308ecd0b4cd3377671/modules/core/04-channel/keeper/packet.go#L442) but I'm unsure if the same semantics would apply).

When implementing the new channel ordering, we will need to modify this code (already specified in the spec). We will likely need to do some grep-ing to look into all switches/conditionals on channel ordering. Core IBC isn't too well setup abstraction wise for new channel types. It's somewhat unclear how many different ordering types will ever exist

Reply:
I'll fix this up today. Better to keep things future proof as much as possible.